test/plugin/base_test.rb

```diff
@@ -605,6 +605,45 @@ def test_sanitize_tags_from_container_name_with_require_valid_tags_false
     end
   end
 
+  def test_configure_split_logs_by_tag
+    setup_gce_metadata_stubs
+    {
+      APPLICATION_DEFAULT_CONFIG => true,
+      DISABLE_SPLIT_LOGS_BY_TAG_CONFIG => false
+    }.each do |(config, split_logs_by_tag)|
+      d = create_driver(config)
+      assert_equal split_logs_by_tag,
+                   d.instance.instance_variable_get(:@split_logs_by_tag)
+    end
+  end
+
+  def test_split_logs_by_tag
+    setup_gce_metadata_stubs
+    log_entries_count = 5
+    params = COMPUTE_PARAMS.reject { |k, _| k == :log_name }
+    [
+      [APPLICATION_DEFAULT_CONFIG, log_entries_count],
+      [DISABLE_SPLIT_LOGS_BY_TAG_CONFIG, 1]
+    ].each do |(config, requests_count)|
+      setup_prometheus
+      setup_logging_stubs do
+        @logs_sent = []
+        d = create_driver(config + PROMETHEUS_ENABLE_CONFIG, 'test', true)
+        log_entries_count.times do |i|
+          d.emit("tag#{i}", 'message' => log_entry(i))
+        end
+        d.run
+        verify_log_entries(log_entries_count, params, 'textPayload')
+        assert_prometheus_metric_value(:stackdriver_successful_requests_count,
```

Comment:
Do we even need to assert these prometheus counts here? Or are you verifying that the requests were sent successfully?

Reply:
Yeah, I added a comment to clarify:
```
        # Verify the number of requests is different based on whether the
        # 'split_logs_by_tag' flag is enabled.
```